BeautifulSoup emits DeprecationWarning

### Software versions

Python version        :  3.11.11 | packaged by conda-forge | (main, Dec  5 2024, 14:17:24) [GCC 13.3.0]
IPython version       :  8.32.0
Tornado version       :  6.4.2
NumPy version         :  2.2.2
Bokeh version         :  3.7.0.dev5+40.ge9ddf9e5.dirty
BokehJS static path   :  /home/marcogorelli/bokeh-dev/src/bokeh/server/static
node.js version       :  v20.18.1
npm version           :  10.8.2
jupyter_bokeh version :  (not installed)
Operating system      :  Linux-5.15.167.4-microsoft-standard-WSL2-x86_64-with-glibc2.35


### Browser name and version

_No response_

### Jupyter notebook / Jupyter Lab version

_No response_

### Expected behavior

No deprecation warning when running tests

### Observed behavior

```
E       DeprecationWarning: Call to deprecated method findAll. (Replaced by find_all) -- Deprecated since version 4.0.0.
```

### Example code


Here's a CI job that shows the warning https://github.com/bokeh/bokeh/actions/runs/13119727091/job/36602824277?pr=14169

### Stack traceback or browser console output

_No response_

### Screenshots

_No response_

--- a/tests/unit/bokeh/embed/test_server__embed.py
+++ b/tests/unit/bokeh/embed/test_server__embed.py
@@ -66,7 +66,7 @@ class TestServerDocument:
         assert 'bokeh-app-path=/foo/bar/sliders' in r
         assert 'bokeh-absolute-url=http://localhost:8081/foo/bar/sliders' in r
         html = bs4.BeautifulSoup(r, "html.parser")
-        scripts = html.findAll(name='script')
+        scripts = html.find_all(name='script')
         assert len(scripts) == 1
         script = scripts[0]
         attrs = script.attrs
@@ -80,7 +80,7 @@ class TestServerDocument:
         r = bes.server_document(arguments=dict(foo=10))
         assert 'foo=10' in r
         html = bs4.BeautifulSoup(r, "html.parser")
-        scripts = html.findAll(name='script')
+        scripts = html.find_all(name='script')
         assert len(scripts) == 1
         script = scripts[0]
         attrs = script.attrs
@@ -95,7 +95,7 @@ class TestServerDocument:
         assert 'bokeh-app-path=/foo/bar/sliders' in r
         assert 'bokeh-absolute-url=http://localhost:8081/foo/bar/sliders' in r
         html = bs4.BeautifulSoup(r, "html.parser")
-        scripts = html.findAll(name='script')
+        scripts = html.find_all(name='script')
         assert len(scripts) == 1
         script = scripts[0]
         attrs = script.attrs
@@ -109,7 +109,7 @@ class TestServerDocument:
         r = bes.server_document(url=url, relative_urls=True)
         assert 'bokeh-app-path=/foo/bar/sliders' in r
         html = bs4.BeautifulSoup(r, "html.parser")
-        scripts = html.findAll(name='script')
+        scripts = html.find_all(name='script')
         assert len(scripts) == 1
         script = scripts[0]
         attrs = script.attrs
@@ -137,7 +137,7 @@ class TestServerSession:
         url = "http://localhost:5006"
         r = bes.server_session(test_plot, session_id='fakesession')
         html = bs4.BeautifulSoup(r, "html.parser")
-        scripts = html.findAll(name='script')
+        scripts = html.find_all(name='script')
         assert len(scripts) == 1
         script = scripts[0]
         attrs = script.attrs
@@ -165,7 +165,7 @@ class TestServerSession:
         url = "http://localhost:5006"
         r = bes.server_session(None, session_id='fakesession')
         html = bs4.BeautifulSoup(r, "html.parser")
-        scripts = html.findAll(name='script')
+        scripts = html.find_all(name='script')
         assert len(scripts) == 1
         script = scripts[0]
         attrs = script.attrs
@@ -179,7 +179,7 @@ class TestServerSession:
         url = "http://localhost:5006"
         r = bes.server_session(test_plot, session_id='fakesession')
         html = bs4.BeautifulSoup(r, "html.parser")
-        scripts = html.findAll(name='script')
+        scripts = html.find_all(name='script')
         assert len(scripts) == 1
         script = scripts[0]
         attrs = script.attrs

--- a/tests/unit/bokeh/test_resources.py
+++ b/tests/unit/bokeh/test_resources.py
@@ -356,7 +356,7 @@ class TestResources:
         r.components.remove("bokeh-mathjax")
         out = r.render_js()
         html = bs4.BeautifulSoup(out, "html.parser")
-        scripts = html.findAll(name='script')
+        scripts = html.find_all(name='script')
         for script in scripts:
             if "src" not in script.attrs:
                 continue
@@ -369,7 +369,7 @@ class TestResources:
         monkeypatch.setattr(resources, "__version__", v)
         out = resources.CDN.render_js()
         html = bs4.BeautifulSoup(out, "html.parser")
-        scripts = html.findAll(name='script')
+        scripts = html.find_all(name='script')
         for script in scripts:
             assert "crossorigin" not in script.attrs
             assert "integrity" not in script.attrs
@@ -382,7 +382,7 @@ class TestResources:
         r.components.remove("bokeh-mathjax")
         out = r.render_js()
         html = bs4.BeautifulSoup(out, "html.parser")
-        scripts = html.findAll(name='script')
+        scripts = html.find_all(name='script')
         for script in scripts:
             if "src" not in script.attrs:
                 continue
@@ -395,7 +395,7 @@ class TestResources:
         monkeypatch.setattr(resources, "__version__", v)
         out = resources.INLINE.render_js()
         html = bs4.BeautifulSoup(out, "html.parser")
-        scripts = html.findAll(name='script')
+        scripts = html.find_all(name='script')
         for script in scripts:
             assert "crossorigin" not in script.attrs
             assert "integrity" not in script.attrs

